src/taskgraph/transforms/from_deps.py

```diff
@@ -206,6 +206,7 @@ def from_deps(config, tasks):
             )
 
             primary_dep = [dep for dep in group if dep.kind == primary_kind][0]
+            new_task["attributes"]["primary-dependency"] = primary_dep
```

Comment:
Should we anticipate issues from storing an object as an attribute rather than a string? Is there precedent for this?

Reply:
There was precedent for it until it was removed in https://phabricator.services.mozilla.com/D193079 :laughing: 

I admit it's a bit weird. Maybe we can just store the primary dependency label, and look it up by that instead, come to think of it?

Reply:
Yes, that works, and is much less weird and memory consuming! It arguably makes this a non-breaking change now, too? (I guess there's other ways that `primary-kind-dependency` might be set other than `from_deps` though...)